pkg/services/ngalert/notifier/receivers.go

```diff
@@ -0,0 +1,214 @@
+package notifier
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	maxTestReceiversWorkers = 10
+)
+
+type TestReceiversResult struct {
+	Receivers []TestReceiverResult
+	NotifedAt time.Time
+}
+
+type TestReceiverResult struct {
+	Name    string
+	Configs []TestReceiverConfigResult
+}
+
+type TestReceiverConfigResult struct {
+	Name   string
+	UID    string
+	Status string
+	Error  error
+}
+
+type InvalidReceiverError struct {
+	Receiver *apimodels.PostableGrafanaReceiver
+	Err      error
+}
+
+func (e InvalidReceiverError) Error() string {
+	return fmt.Sprintf("the receiver %s is invalid: %s", e.Receiver.Name, e.Err)
+}
+
+type ReceiverTimeoutError struct {
+	Receiver *apimodels.PostableGrafanaReceiver
+	Err      error
+}
+
+func (e ReceiverTimeoutError) Error() string {
+	return fmt.Sprintf("the receiver timed out: %s", e.Err)
+}
+
+func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestReceiversConfigParams) (*TestReceiversResult, error) {
+	// now represents the start time of the test
+	now := time.Now()
+	testAlert := &types.Alert{
+		Alert: model.Alert{
+			Labels: model.LabelSet{
+				model.LabelName("alertname"): "TestAlertAlwaysFiring",
+				model.LabelName("instance"):  "Grafana",
+			},
+			Annotations: model.LabelSet{},
```

Comment:
Can some annotations be added here?

My think is that this sort of acts as a mock for users, so by having some annotations here, a user can get a better picture of what the contact point + template will look like.